src/v/utils/log_hist.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "utils/log_hist.h"
+
+template<
+  typename duration_t,
+  int number_of_buckets,
+  uint64_t first_bucket_upper_bound>
```

Comment:
what's the benefit of having these two constants be template parameters? as far as i can tell they aren't involved in much compile time optimizations. but maybe i'm missing this?

Reply:
Number of buckets is so we can use an array instead of a vector and as a compile-time constant for the upper bound in the `record` function. The first bucket upper bound is used to compute `first_bucket_clz` at compile time in the `record` function.

Reply:
ahh makes sense.